### PR TITLE
Supermassive codegen fixes

### PR DESCRIPTION
**Pull request by vejrj**

### CODE REVIEW

File: packages/supermassive/src/codegen/__tests__/index.test.ts [hunk 1]
```diff
@@ -49,27 +49,27 @@ describe(generateTS, () => {
...
-            avatar: AvatarModel | null;
-            avatarRequired: AvatarModel;
+            readonly __typename: \\"User\\";
+            readonly id: string;
+            readonly name: string | null;
+            readonly messagesWithAnswersNonRequired: ((MessageModel | null)[] | null)[] | null;
+            readonly messagesWithAnswersRequired: ((MessageModel | null)[] | null)[];
+            readonly messagesWithAnswersAllRequired: MessageModel[][];
+            readonly messagesNonRequired: (MessageModel | null)[] | null;
+            readonly messagesWithArrayRequired: (MessageModel | null)[];
```
Comment (freiksenet) [inline packages/supermassive/src/codegen/__tests__/index.test.ts:66]: Arrays should be ReadonlyArray types.

File: packages/supermassive/src/codegen/__tests__/index.test.ts [hunk 2]
```diff
@@ -139,9 +139,9 @@ describe(generateTS, () => {
             export type id = (model: UserModel, args: {}, context: unknown, info: ResolveInfo) => PromiseOrValue<string>;
         }
         export declare namespace Subscription {
-            export type userUpdated<A = unknown> = ((model: unknown, args: {}, context: unknown, info: ResolveInfo) => AsyncIterator<UserModel>) | {
+            export type userUpdated<A = unknown> = {
```
Comment (freiksenet) [inline packages/supermassive/src/codegen/__tests__/index.test.ts:142]: Let me test how to make this A work better and then we can merge.

**Review by freiksenet — APPROVED** [review]

**Review by freiksenet — APPROVED** [review]

**Comment by freiksenet** [comment]
Let's do the readonly array fix and gimme an hour to test one thing.

File: packages/supermassive/src/codegen/__tests__/index.test.ts (hunk 1 above)
Comment (vejrj) [reply packages/supermassive/src/codegen/__tests__/index.test.ts]: done

**Review by freiksenet — APPROVED** [review]

**Pull request merged**